Dockerfile

```diff
@@ -22,7 +22,7 @@ WORKDIR /go/src/github.com/docker/cli
 FROM build-base-alpine AS build-alpine
 ARG TARGETPLATFORM
 # gcc is installed for libgcc only
-RUN xx-apk add --no-cache musl-dev gcc
+RUN xx-apk add --no-cache musl-dev libgcc
```

Comment:
Interesting; this is causing failures on arm/v6 and arm/v7; it looks like we're explicitly disabling CGO (`CGO_ENABLED=0`), but then try to compile with `-buildmode=pie` - seems like something is off somewhere.


```
 > [build-plugins 1/1] RUN --mount=ro --mount=type=cache,target=/root/.cache     xx-go --wrap &&     TARGET=/out ./scripts/build/plugins e2e/cli-plugins/plugins/*:
0.127 Building static docker-helloworld
0.127 + CGO_ENABLED=0
0.127 + GO111MODULE=auto
0.127 + go build -o /out/plugins-linux-arm/docker-helloworld -tags ' osusergo' -ldflags ' -X "github.com/docker/cli/cli/version.GitCommit=5c123b1" -X "github.com/docker/cli/cli/version.BuildTime=2024-09-02T13:52:17Z" -X "github.com/docker/cli/cli/version.Version=pr-5387" -extldflags -static' -buildmode=pie github.com/docker/cli/cli-plugins/examples/helloworld
0.135 -buildmode=pie requires external (cgo) linking, but cgo is not enabled
```

Reply:
Right, so the `.variables` sets `CGO_ENABLED=1` on arm; https://github.com/docker/cli/blob/b0c41b78d8c80c5f8faa736cf3c74dc31092e443/scripts/build/.variables#L57-L68

And if enabled, it sets `-buildmode=pie`; https://github.com/docker/cli/blob/b0c41b78d8c80c5f8faa736cf3c74dc31092e443/scripts/build/.variables#L79-L88

So that looks to be conflicting with the hard-coded `CGO_ENABLED=0` in this script.

Reply:
Interestingly; doesn't fail on other PRs - wondering now what's causing that issue;

```
#18 [build-plugins 1/1] RUN --mount=ro --mount=type=cache,target=/root/.cache     xx-go --wrap &&     TARGET=/out ./scripts/build/plugins e2e/cli-plugins/plugins/*
#18 0.126 Building static docker-helloworld
#18 0.126 + CGO_ENABLED=0
#18 0.126 + GO111MODULE=auto
#18 0.126 + go build -o /out/plugins-linux-arm/docker-helloworld -tags ' osusergo' -ldflags ' -X "github.com/docker/cli/cli/version.GitCommit=0cebca4" -X "github.com/docker/cli/cli/version.BuildTime=2024-09-02T14:50:36Z" -X "github.com/docker/cli/cli/version.Version=pr-5389" -extldflags -static' -buildmode=pie github.com/docker/cli/cli-plugins/examples/helloworld
#18 19.75 # github.com/docker/cli/cli-plugins/examples/helloworld
#18 19.75 loadinternal: cannot find runtime/cgo
#18 19.85 Building static docker-badmeta
#18 19.85 + CGO_ENABLED=0
#18 19.85 + GO111MODULE=auto
```